### ISSUE TITLE
Representative links become broken after swiping

### ISSUE THREAD

**Issue by is4kov** [glidejs/glide#392]
On swiping, `Anchors.attach()` and `Anchors.detach()` methods are manipulating with `href` of links, but in case the link is representative (has no `href`) it becomes broken after this, as there is no check of `href` existance, in result such link has `data-href="null" href="null"`.

Seems like the best way to fix this is to renew `Anchors`, using more precise selector (`a[href]` instead of `a`), and do this right on swipe start.

Can send PR with fix.

**Comment by nicospoke** [comment]
I have a carousel with glideJS which I destroy when viewport is desktop size. I then mount the slider again on mobile. 

Can you post your code so I can be inspired to fix my issue?

**Comment by andrewgadziksonos** [comment]
I think the issue is here https://github.com/glidejs/glide/blob/e894ec359668df58a41568619d8ed33cc941916c/src/components/anchors.js#L146

When using a mouse, `swipe.move` seems to be fired multiple times before `swipe.end` is fired, which causes `detach()` to be executed multiple times before `attach()` is executed.  

It's on that second `detach()` execution where `href` is null and `data-href` is set to null.

I believe the fix is to listen on `swipe-start` to detach href's

**Comment by ericmorand** [comment]
href attribute is mandatory per HTML specifications. If not, then this is not a link:

> The destination of the link(s) is given by the href attribute, which must be present and must contain a valid non-empty URL potentially surrounded by spaces. If the href attribute is absent, then the element does not define a link.

Glide has no reason to support a behaviour that doesn't respect HTML specifications.

**Comment by is4kov** [comment]
> href attribute is mandatory per HTML specifications. If not, then this is not a link

In HTML5 it is not required - [w3c](https://www.w3.org/TR/2011/WD-html5-20110113/links.html#links-created-by-a-and-area-elements)

**Comment by ericmorand** [comment]
> The href attribute on a and area elements is not required; when those elements do not have href attributes they do *not* create hyperlinks.

This is not an hyperlink then. Which means that the `a` tag is treated as a non-typed inline element. Which makes changing href later a non-issue since this is not a link to begin with.

We raises another question: why using an anchor tag to represent something that is not an hyperlink?

**Comment by BenJBailey** [comment]
@nicospoke, I was also experiencing this issue because I was attempting to destroy and recreate Glide instances on the same element on window.resize. Turns out I was calling [destroy()](https://glidejs.com/docs/api/#destroy-) incorrectly, but I also realized that wasn't the ideal solution:

I didn't know that [update()](https://glidejs.com/docs/api/#update-settings-) was a possibility. I opted to use [update()](https://glidejs.com/docs/api/#update-settings-) instead. That offloaded the destroy and recreate logic to Glide.

I did something similar to this

```
var glide = new Glide('.selector', { perView:5 });

window.addEventListener("resize", function() {
  if (document.documentElement.clientWidth < 768) {
     glide.update({ preView: 3});
  } else {
     glide.update({ preView: 5});
  }
});
```

**Comment by BenJBailey** [comment]
I should also say: hrefs remain intact if you opt to use [update()](https://glidejs.com/docs/api/#update-settings-).

**Comment by jedrzejchalubek** [comment]
Jugglery links urls between `href` and `data-href` attributes has been removed with `v3.5.0`
